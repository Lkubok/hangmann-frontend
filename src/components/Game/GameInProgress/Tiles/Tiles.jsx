import React, { Component } from "react";
import { connect } from "react-redux";
import * as selectors from "../../../../reducers/selectors";
import PropTypes from "prop-types";
import "./Tiles.scss";

export class Tiles extends Component {
  renderTiles = () => {
    const { lettersToGuess } = this.props;
    return lettersToGuess.map((el, index) => (
      <div
        key={index}
        className={
          el === "encoded"
            ? "tile tile-to-guess"
            : [
                " ",
                "?",
                "!",
                ",",
                ".",
                ":",
                ";",
                "-",
                "+",
                "'",
                "`",
                "1",
                "2",
                "3",
                "4",
                "5",
                "6",
                "7",
                "8",
                "9",
                "0"
              ].includes(el)
            ? "tile tile-space"
            : "tile tile-guessed"
        }
      >
        {el === "encoded" ? "" : el}
      </div>
    ));
  };
  render() {
    return <div className="letter-box">{this.renderTiles()}</div>;
  }
}

const mapStateToProps = state => ({
  lettersToGuess: selectors.getLettersToGuess(state),
  typed: selectors.getTypedLetters(state)
});

const mapDispatchToProps = {};

Tiles.propTypes = {
  lettersToGuess: PropTypes.array,
  typed: PropTypes.array
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Tiles);
