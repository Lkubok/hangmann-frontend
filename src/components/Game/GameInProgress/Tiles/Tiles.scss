@import "../../../../scss/mixins.scss";
@import "../../../../scss/variables.scss";

$height: 80;
$width: 50;

.letter-box {
  @include flex-cc-row;
}

.tile {
  @media all and (max-width: 768px) {
    height: 35px;
  width: 25px;
  font-size: 1em;
  font-weight: 500;
  margin: 2px


  }
  @include flex-cc-row;
  height: 60px;
  width: 40px;
  margin: 10px;
  border: 1px solid $dark-color;
  border-radius: 15px;
  font-weight: 700;
  font-size: 1.8em;
  text-transform: capitalize;
  &-to-guess {
    background-color: $white-color;
  }
  &-guessed {
    background-color: $primary-color;
  }
  &-space {
    background-color: $dark-color;
    border-color: $dark-color-light;
  }
}


