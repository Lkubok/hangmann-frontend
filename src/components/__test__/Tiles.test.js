import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Tiles } from "../Game/GameInProgress/Tiles/Tiles";
Enzyme.configure({ adapter: new Adapter() });

describe("Tiles component", () => {
  let props;
  beforeEach(() => {
    props = {
      lettersToGuess: ["a", "b"]
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<Tiles {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
