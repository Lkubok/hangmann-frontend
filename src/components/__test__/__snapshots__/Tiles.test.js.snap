// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiles component Should match snapshot 1`] = `
<div
  className="letter-box"
>
  <div
    className="tile tile-guessed"
    key="0"
  >
    a
  </div>
  <div
    className="tile tile-guessed"
    key="1"
  >
    b
  </div>
</div>
`;
